rbc/rbc.go

```diff
@@ -1,100 +1,483 @@
 package rbc
 
 import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
 	"github.com/DE-labtory/cleisthenes"
 	"github.com/DE-labtory/cleisthenes/pb"
+	"github.com/DE-labtory/cleisthenes/rbc/merkletree"
+	"github.com/DE-labtory/iLogger"
+	"github.com/golang/protobuf/ptypes"
 	"github.com/klauspost/reedsolomon"
+	"github.com/syndtr/goleveldb/leveldb/errors"
 )
 
-type RBC struct {
+var ErrInvalidRBCType = errors.New("Invalid RBC message type")
 
+type RBC struct {
 	// number of network nodes
 	n int
 
 	// number of byzantine nodes which can tolerate
 	f int
 
+	// owner of rbc instance (node)
+	owner cleisthenes.Member
+
+	// proposerId is the ID of proposing node
 	proposer cleisthenes.Member
 
 	// Erasure coding using reed-solomon method
 	enc reedsolomon.Encoder
 
-	// Broadcast message after when rbc receive message
-	messages []*pb.Message
+	// output of RBC
+	value []byte
+
+	// length of original data
+	contentLength uint64
+
+	// number of sharded data and parity
+	// data : N - F, parity : F
+	numDataShards, numParityShards int
 
 	// Request of other rbcs
-	valueReqRepo cleisthenes.RequestRepository
 	echoReqRepo  cleisthenes.RequestRepository
 	readyReqRepo cleisthenes.RequestRepository
 
+	valReceived, echoSent, readySent bool
+
 	// internal channels to communicate with other components
-	closeChan   chan struct{}
-	messageChan chan InnerMessage
-	requestChan chan InnerRequest
+	closeChan chan struct{}
+	reqChan   chan request
+	inputChan chan InputMessage
+
+	broadcaster cleisthenes.Broadcaster
+}
+
+func New(n, f int, owner, proposer cleisthenes.Member, broadcaster cleisthenes.Broadcaster) *RBC {
+	numParityShards := f
+	numDataShards := n - numParityShards
+
+	enc, err := reedsolomon.New(numDataShards, numParityShards)
+	if err != nil {
+		panic(err)
+	}
+
+	echoReqRepo, err := NewEchoReqRepository()
+	if err != nil {
+		panic(err)
+	}
+
+	readyReqRepo, err := NewReadyReqRepository()
+	if err != nil {
+		panic(err)
+	}
+
+	rbc := &RBC{
+		n:               n,
+		f:               f,
+		owner:           owner,
+		proposer:        proposer,
+		enc:             enc,
+		numDataShards:   numDataShards,
+		numParityShards: numParityShards,
+		echoReqRepo:     echoReqRepo,
+		readyReqRepo:    readyReqRepo,
+		closeChan:       make(chan struct{}),
+		reqChan:         make(chan request),
+		inputChan:       make(chan InputMessage),
+		broadcaster:     broadcaster,
+	}
+	go rbc.run()
+	return rbc
+}
+
+func (rbc *RBC) distributeMessage(proposer cleisthenes.Member, reqs []cleisthenes.Request) error {
+	if len(reqs) != rbc.n {
+		return fmt.Errorf("invalid message count - expected : %d, got : %d", rbc.n, len(reqs))
```

Comment:
plz unify creating error function 

1. fmt.Errorf
2. errors.New <- this used more often

Reply:
If you want to use formatted message, use `errors.New(fmt.Sprintf)`